### PR TITLE
Convert the export to use SpanData objects, rather than protobufs

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java [hunk 1]
```diff
@@ -143,44 +156,44 @@ private Adapter() {}
    *
    * @param attributes the span attributes
    * @return a collection of Jaeger key values
-   * @see #toKeyValue(String, AttributeValue)
+   * @see #toKeyValue(String, io.opentelemetry.trace.AttributeValue)
```
Comment (jkwatson) [inline exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java]: import?

File: exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java [hunk 2]
```diff
@@ -146,9 +160,9 @@ private Adapter() {}
    * @see #toKeyValue(String, AttributeValue)
    */
   @VisibleForTesting
-  static Collection<Model.KeyValue> toKeyValues(Span.Attributes attributes) {
-    ArrayList<Model.KeyValue> tags = new ArrayList<>(attributes.getAttributeMapCount());
-    for (Map.Entry<String, AttributeValue> entry : attributes.getAttributeMapMap().entrySet()) {
+  static Collection<Model.KeyValue> toKeyValues(Map<String, AttributeValue> attributes) {
+    ArrayList<Model.KeyValue> tags = new ArrayList<>(attributes.size());
```
Comment (breedx-nr) [inline exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java:164]: `List` on lefthand side?

File: opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/errorreporting/ErrorReportingTest.java [hunk 3]
```diff
@@ -59,9 +61,9 @@ public void testSimpleError() {
 
     assertNull(tracer.scopeManager().activeSpan());
 
-    List<io.opentelemetry.proto.trace.v1.Span> spans = exporter.getFinishedSpanItems();
+    List<SpanData> spans = exporter.getFinishedSpanItems();
     assertEquals(spans.size(), 1);
-    assertEquals(spans.get(0).getStatus().getCode(), Status.UNKNOWN.getCanonicalCode().value());
+    assertEquals(spans.get(0).getStatus().getCanonicalCode(), Status.UNKNOWN.getCanonicalCode());
```
Comment (breedx-nr) [inline opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/errorreporting/ErrorReportingTest.java:66]: put expected on left

File: sdk/src/main/java/io/opentelemetry/sdk/trace/ReadableSpanAdapter.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import io.opentelemetry.sdk.trace.export.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanData.Event;
+import io.opentelemetry.sdk.trace.export.SpanData.TimedEvent;
+import io.opentelemetry.sdk.trace.export.SpanData.Timestamp;
+import io.opentelemetry.trace.SpanId;
+import java.util.ArrayList;
+import java.util.List;
+
+/** An adapter that can convert a ReadableSpan into a SpanData. */
+public class ReadableSpanAdapter {
```
Comment (breedx-nr) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/ReadableSpanAdapter.java:29]: needs `@since`

**Review by breedx-nr — APPROVED** [review]
Looking good.  Let's squash and PR against upstream.

**Pull request merged**